tests/integration/test_sqs.py

```diff
@@ -1398,16 +1400,631 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
+            QueueUrl=queue_url, VisibilityTimeout=0, MessageAttributeNames=["All"]
+        )
+        receive_result = sqs_client.receive_message(
+            QueueUrl=queue_url, MessageAttributeNames=["All"]
+        )
+        assert receive_result["Messages"][0]["MessageAttributes"] == attributes
+
+    def test_send_message_with_invalid_string_attributes(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        # String Attributes must not contain non-printable characters
+        # See: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+        invalid_attribute = {
+            "attr1": {"StringValue": "Invalid-{},{}".format(chr(8), chr(11)), "DataType": "String"}
+        }
+
+        with pytest.raises(Exception) as e:
+            sqs_client.send_message(
+                QueueUrl=queue_url, MessageBody="test", MessageAttributes=invalid_attribute
+            )
+        e.match("Invalid")
+
+    def test_send_message_with_invalid_payload_characters(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        invalid_message_body = "Invalid-{}-{}-{}-{}".format(chr(0), chr(8), chr(19), chr(65535))
+
+        with pytest.raises(Exception) as e:
+            sqs_client.send_message(QueueUrl=queue_url, MessageBody=invalid_message_body)
+        e.match("InvalidMessageContents")
+
+    def test_dead_letter_queue_config(self, sqs_client, sqs_create_queue):
+        # TODO: not tested against AWS
+        queue_name = "queue-{}".format(short_uid())
+        dead_letter_queue_name = "dead_letter_queue-{}".format(short_uid())
+
+        dl_queue_url = sqs_create_queue(QueueName=dead_letter_queue_name)
+        url_parts = dl_queue_url.split("/")
+        region = os.environ.get("AWS_DEFAULT_REGION") or TEST_REGION
+        dl_target_arn = "arn:aws:sqs:{}:{}:{}".format(
+            region, url_parts[len(url_parts) - 2], url_parts[len(url_parts) - 1]
+        )
+
+        conf = {"deadLetterTargetArn": dl_target_arn, "maxReceiveCount": 50}
+        attributes = {"RedrivePolicy": json.dumps(conf)}
+
+        queue_url = sqs_create_queue(QueueName=queue_name, Attributes=attributes)
+
+        assert queue_url
+
+    def test_dead_letter_queue_execution(
+        self, sqs_client, sqs_create_queue, lambda_client, create_lambda_function
+    ):
+
+        # TODO: lambda creation does not work when testing against AWS
+        queue_name = "queue-{}".format(short_uid())
+        dead_letter_queue_name = "dl-queue-{}".format(short_uid())
+        dl_queue_url = sqs_create_queue(QueueName=dead_letter_queue_name)
+
+        # create arn
+        url_parts = dl_queue_url.split("/")
+        region = os.environ.get("AWS_DEFAULT_REGION") or TEST_REGION
+        dl_target_arn = "arn:aws:sqs:{}:{}:{}".format(
```

Comment:
for arn creation we can maybe re-use `sqs_queue_arn` from the `aws_stack` utils.

Reply:
The problem was that `sqs_queue_arn` always returned the local account id even whien testing against AWS because it was not explicitly supplied. I was unsure of how to properly access the existing ID provided via the profile as that the CI will properly pick it up as well